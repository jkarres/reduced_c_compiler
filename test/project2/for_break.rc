
function : void main() {
    int i, j;

    for (i = 0; i < 10; ++i) {

        for (j = 0; j < 10; ++j) {

            int product = i * j;

            if (product > 50) {
                cout << "These numbers are too big.  I'm breaking from the j loop!" << endl;
                break;
            }

            cout << i << " * " << j << " is " << product << "." << endl;

        }

        cout << endl;

    }

}
