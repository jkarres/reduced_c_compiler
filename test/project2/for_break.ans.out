0 * 0 is 0.
0 * 1 is 0.
0 * 2 is 0.
0 * 3 is 0.
0 * 4 is 0.
0 * 5 is 0.
0 * 6 is 0.
0 * 7 is 0.
0 * 8 is 0.
0 * 9 is 0.

1 * 0 is 0.
1 * 1 is 1.
1 * 2 is 2.
1 * 3 is 3.
1 * 4 is 4.
1 * 5 is 5.
1 * 6 is 6.
1 * 7 is 7.
1 * 8 is 8.
1 * 9 is 9.

2 * 0 is 0.
2 * 1 is 2.
2 * 2 is 4.
2 * 3 is 6.
2 * 4 is 8.
2 * 5 is 10.
2 * 6 is 12.
2 * 7 is 14.
2 * 8 is 16.
2 * 9 is 18.

3 * 0 is 0.
3 * 1 is 3.
3 * 2 is 6.
3 * 3 is 9.
3 * 4 is 12.
3 * 5 is 15.
3 * 6 is 18.
3 * 7 is 21.
3 * 8 is 24.
3 * 9 is 27.

4 * 0 is 0.
4 * 1 is 4.
4 * 2 is 8.
4 * 3 is 12.
4 * 4 is 16.
4 * 5 is 20.
4 * 6 is 24.
4 * 7 is 28.
4 * 8 is 32.
4 * 9 is 36.

5 * 0 is 0.
5 * 1 is 5.
5 * 2 is 10.
5 * 3 is 15.
5 * 4 is 20.
5 * 5 is 25.
5 * 6 is 30.
5 * 7 is 35.
5 * 8 is 40.
5 * 9 is 45.

6 * 0 is 0.
6 * 1 is 6.
6 * 2 is 12.
6 * 3 is 18.
6 * 4 is 24.
6 * 5 is 30.
6 * 6 is 36.
6 * 7 is 42.
6 * 8 is 48.
These numbers are too big.  I'm breaking from the j loop!

7 * 0 is 0.
7 * 1 is 7.
7 * 2 is 14.
7 * 3 is 21.
7 * 4 is 28.
7 * 5 is 35.
7 * 6 is 42.
7 * 7 is 49.
These numbers are too big.  I'm breaking from the j loop!

8 * 0 is 0.
8 * 1 is 8.
8 * 2 is 16.
8 * 3 is 24.
8 * 4 is 32.
8 * 5 is 40.
8 * 6 is 48.
These numbers are too big.  I'm breaking from the j loop!

9 * 0 is 0.
9 * 1 is 9.
9 * 2 is 18.
9 * 3 is 27.
9 * 4 is 36.
9 * 5 is 45.
These numbers are too big.  I'm breaking from the j loop!

